app/views/chargeback/_cb_rate_edit_table.html.haml

```diff
@@ -18,13 +18,14 @@
       %th= _("Fixed")
       %th= _("Variable")
   %tbody
-    - @edit[:new][:details].sort_by { |rd| [rd[:group], rd[:description], rd[:sub_metric].to_s] }.each do |detail|
+    - @edit[:new][:details].sort_by { |rd| [rd[:group], rd[:description], rd[:sub_metric].to_s] }.each_with_index do |detail, detail_index|
       - @cur_group = detail[:group] if @cur_group.nil?
       - if @cur_group != detail[:group]
         - @cur_group = detail[:group]
         %tr
           %td{:colspan => "10", :style => "background-color: #f5f5f5;"} &nbsp;
-      - detail_index = @edit[:new][:tiers].index { |x| x.detect { |tier_hash| tier_hash['chargeback_rate_detail_id'] == detail[:id] }.present? }
+      - if params[:pressed] == "chargeback_rates_edit"
```

Comment:
This doesn't sound like the best place for this kind of condition.

Also, this means `detail_index` will be `nil` for all the other cases...


So.. all the code like `{:id => "rate_detail_row_#{detail_index}_0"}` has a bug now.

Please make sure `detail_index` always has a reasonable value.
(And ideally, move the `params[:pressed]` condition to controller code.)

Reply:
`detail_index` will not be `nil ` because of https://github.com/ManageIQ/manageiq-ui-classic/pull/3129/files#diff-9e228e37bdb4a0dfb3c30cffd92d3981R21 
I have tested that it always will be set, and properly. `detail_index` is just changed `if params[:pressed] == "chargeback_rates_edit"`
Or am I missing something?
This change is made because in this haml... this haml is common for editing and displaying the rates. For just displaying,  using `detail_index` just as it is, works perfectly. When editing, we need it just like this: https://github.com/ManageIQ/manageiq-ui-classic/pull/3129/files#diff-9e228e37bdb4a0dfb3c30cffd92d3981L27